browser/mocha/replace.js

```diff
@@ -32,87 +32,51 @@ extendInterfaces('replace', function(context, teardown) {
-              while (replacements.hasOwnProperty(currentTagName)) {
-                currentTagName = replacements[currentTagName];
-              }
-
-              // if we have not changed this element, copy it over
-              if (currentTagName === originalTagName) {
-                instanceNode = document.importNode(templateNode);
-
-              } else {
-                // create the new node
-                instanceNode = document.createElement(currentTagName);
-
-                var numAttributes = templateNode.attributes.length;
-                // For all attributes in the original node..
-                for (var index=0; index<numAttributes; ++index) {
-                  // Set that attribute on the new node:
-                  instanceNode.setAttribute(templateNode.attributes[index].name,
-                      templateNode.attributes[index].value);
-                }
-              }
-
-            } else {
-              // if it is not an element node, simply import it.
-              instanceNode = document.importNode(templateNode);
-            }
-
-            if (instanceParent) {
-              // Polymer's shady dom implementation goes through the insertion
-              // points and checks their parents. If the parent of a content
-              // tag has been stamped already, then Polymer.dom has to be aware
-              // of this content tag's parent. Additionally,
-              // Polymer.dom.appendChild does not seem to actually append the
-              // content nodes into the document fragment, so node.appendChild
-              // must also be called to actually insert the node.
-              if (instanceNode.tagName == 'CONTENT') {
-                Polymer.dom(instanceParent).appendChild(instanceNode);
-              }
-
-              instanceParent.appendChild(instanceNode);
-            }
-
-            // traverse down the tree
-            if (templateNode.firstChild) {
-              instanceParent = instanceNode;
-              templateNode = templateNode.firstChild;
-
-            // traverse up
-            } else {
-              // traverse up until you can move laterally
-              while (!templateNode.nextSibling) {
-                // stop traversing up if we are at the top
-                if (templateNode.parentNode === dom) {
-                  return instanceParent;
-                } else {
-                  instanceParent = instanceParent.parentNode;
-                }
+          var origContent = template._content || template.content;
+          var templateClone = document.createElement('template');
+          var content = templateClone.content;
+          var inertDoc = content.ownerDocument;
+
+          // appends children to templateClone
+          for (var n=origContent.firstChild; n; n=n.nextSibling) {
+            // imports node from inertDoc which holds inert nodes.
+            templateClone.content.appendChild(inertDoc.importNode(n, true));
```

Comment:
Do we have a test that tests replace with a template with bindings and/or if or repeat? @kevinpschaaf was saying we would need to copy `_notes` properties into the cloned template nodes.